File: Python/car.py
from account import Account


class Car:
    __id = int
    __license = str
    __driver = Account('', '', '', '')
    _passenger = int(0)

    def __init__(self, license, driver):
        self.license = license
        self.driver = driver

    def printDataCar(self):
        if(self._passenger != 0):
            print('License:', self.license, ' Driver:', self.driver.name)

    def getPassenger(self):
        return self._passenger

    def setPassenger(self, passenger):
        if (passenger != 4):
            print('You must assign 4 passengers')
        self._passenger = passenger

    def getId(self):
        return self.__id

    def setId(self, id):
        self.__id = id

    def getLicense(self):
        return self.__license

    def setLicense(self, license):
        self.__license = license

    def getDriver(self):
        return self.__driver

    def setDriver(self, driver):
        self.__driver = driver

File: TypeScript/Car.ts
import { Account } from "./Account";

export class Car {
    private id: Number;
    private license: String;
    private driver: Account;
    protected passenger: Number;

    constructor( license: String, driver: Account) {
        this.license = license;
        this.driver = driver;
    }

    public printDataCar(): void {
        if( this.passenger != null ) {
            console.log('License: ' + this.license + ' Driver: ' + this.driver.name);
        }
    }

    public get getPassenger() : Number {
        return this.passenger;
    }

    public set setPassenger( passenger: Number ) {
        if( passenger != 4 ) {
            console.error("You must assign 4 passengers");
        }
        this.passenger = passenger;
    }

    public get getId() : Number {
        return this.id;
    }

    public set setId( id: Number ) {
        this.id = id;
    }

    public get getLicense() : String {
        return this.license;
    }

    public set setLicense( license: String ) {
        this.license = license;
    }

    public get getDriver() : Account {
        return this.driver;
    }

    public set setDriver( driver: Account ) {
        this.driver = driver;
    }
}
